Edit Tailwind style section to change links and wording (#11279)
withastro/docs: src/content/docs/en/guides/styling.mdx

### Tailwind

> [Tailwind support in Astro is provided by a Vite plugin](https://tailwindcss.com/docs/installation/framework-guides/astro). To install the plugin in Astro  `>=5.2.0`, use the `astro add tailwind` command for your package manager. For earlier versions of Astro, follow the instructions to [install the official Vite plugin][tailwind] manually.

Reviewer: ```suggestion
Tailwind support in Astro is provided by a Vite plugin. [Follow the Tailwind documentation](https://tailwindcss.com/docs/installation/framework-guides/astro) to add this plugin manually to your Astro project.

 For Astro  `>=5.2.0`, the `astro add tailwind` command for your package manager will install this plugin for you. For earlier versions of Astro, follow the instructions to [install the official Vite plugin][tailwind] manually.
```

> [Tailwind support in Astro is provided by a Vite plugin](https://tailwindcss.com/docs/installation/framework-guides/astro). To install the plugin in Astro  `>=5.2.0`, use the `astro add tailwind` command for your package manager. For earlier versions of Astro, follow the instructions to [install the official Vite plugin][tailwind] manually.

Author: ```suggestion
Tailwind support in Astro is provided by a Vite plugin. [Follow the Tailwind documentation](https://tailwindcss.com/docs/installation/framework-guides/astro) to add this plugin to your Astro project using the official method.

If your project uses Astro  `>=5.2.0`, you can instead use the `astro add tailwind` command for your package manager:
```

@sarah11918 This is what I was thinking.

Reviewer: So, I've admittedly been waiting a while on this one because I wasn't sure how soon we were removing the old plugin and therefore when we'd basically only be supporting Tailwind 4. 😅 

I'm realizing now that it seems many are still using (wanting to use) Tailwind 3 and these instructions are exclusively for Tailwind 4. In such a case, the v4 docs for Tailwind to install the (deprecated) `@astrojs/tailwind` from NPM (minus using `astro add`) would be appropriate.

So just catching up on why I've been unresponsive here, but now that I've had confirmation that the plan is to remove the integration entirely next week, I think we can use this PR to *also* help people use Tailwind 3, if that's their preference. Will probably make a suggestion later today about this!

### Tailwind

> You can add Tailwind to support to your project with a CLI command, or install `@astrojs/tailwind` manually for [legacy Tailwind 3 support](#legacy-tailwind-3-support).

Reviewer: ```suggestion
You can add Tailwind 4 to support to your project with a CLI command, or install `@astrojs/tailwind` manually for [legacy Tailwind 3 support](#legacy-tailwind-3-support).
```

### Tailwind

> You can add Tailwind 4 to support to your project with a CLI command, or install legacy dependencies manually for [legacy Tailwind 3 support](#legacy-tailwind-3-support).

Reviewer 2: ```suggestion
You can add Tailwind 4 support to your project with a CLI command, or install legacy dependencies manually for [legacy Tailwind 3 support](#legacy-tailwind-3-support).
```

---
```

#### Legacy Tailwind 3 support

> To add (or keep) support for Tailwind 3, you will need to have both `tailwindcss@3` and the official Astro Tailwind integration `@astrojs/tailwind` installed. These dependencies are only used for legacy Tailwind 3 compatibility, and are not required for Tailwind 4. You will also need a legacy Tailwind configuration:

Reviewer: ```suggestion
To add (or keep) support for Tailwind 3, you will need to have both `tailwindcss@3` and the official Astro Tailwind integration `@astrojs/tailwind` installed. Installing these dependencies manually is only used for legacy Tailwind 3 compatibility, and is not required for Tailwind 4. You will also need a legacy Tailwind configuration:
```

> To add (or keep) support for Tailwind 3 to an Astro project, you will need to have (or keep) an older version (`v5.1.5`) of the official Astro Tailwind integration `@astrojs/tailwind` installed. This version of the integration will be excluded when running Astro's command to upgrade all dependencies. However, you may see a message in your terminal with instructions for upgrading to Tailwind 4. You will also need a legacy Tailwind configuration:

Reviewer 3: I think we can recommend using v6 since we're not deprecating it

> To add (or keep) support for Tailwind 3 to an Astro project, you will need to have (or keep) an older version (`v5.1.5`) of the official Astro Tailwind integration `@astrojs/tailwind` installed. This version of the integration will be excluded when running Astro's command to upgrade all dependencies. However, you may see a message in your terminal with instructions for upgrading to Tailwind 4. You will also need a legacy Tailwind configuration:

Reviewer 3: > However, you may see a message in your terminal with instructions for upgrading to Tailwind 4

What is this referring to? Is it supposed to be the deprecation NPM package for the day we deprecate?

<Steps>

> 1. Install `v5.1.5` to your project dependencies using your preferred package manager:

Reviewer: ```suggestion
1. Install Tailwind and the Astro Tailwind integration to your project dependencies using your preferred package manager:
```

   <PackageManagerTabs>
     <Fragment slot="npm">
     ```shell

>      npm install @astrojs/tailwind@5.1.5

Reviewer 3: Needs to also install `tailwindcss@3` (there's probably a better version to specify here but you get the idea)

> 1. Install Tailwind and the Astro Tailwind integration to your project dependencies using your preferred package manager:
> 
>    <PackageManagerTabs>
>      <Fragment slot="npm">
>      ```shell
>      npm install @astrojs/tailwind@5.1.5
>      ```
>      </Fragment>
>      <Fragment slot="pnpm">
>      ```shell
>      pnpm add @astrojs/tailwind@5.1.5
>      ```
>      </Fragment>
>      <Fragment slot="yarn">
>      ```shell
>      yarn add @astrojs/tailwind@5.1.5
>      ```
>      </Fragment>
>    </PackageManagerTabs>

Reviewer: ``````suggestion
1. Install Tailwind and the Astro Tailwind integration to your project dependencies using your preferred package manager:

   <PackageManagerTabs>
     <Fragment slot="npm">
     ```shell
     npm install tailwindcss@3 @astrojs/tailwind
     ```
     </Fragment>
     <Fragment slot="pnpm">
     ```shell
     pnpm add tailwindcss@3 @astrojs/tailwind@5.1.5
     ```
     </Fragment>
     <Fragment slot="yarn">
     ```shell
     yarn add tailwindcss@3 @astrojs/tailwind@5.1.5
     ```
     </Fragment>
   </PackageManagerTabs>
``````

2. Import the integration to your `astro.config.mjs` file, and add it to your `integrations[]` array:

>     ```js title="astro.config.mjs" ins={2} ins="sitemap()"

Reviewer: ```suggestion
    ```js title="astro.config.mjs" ins={2} ins="tailwind()"
```

    import { defineConfig } from 'astro/config';

>     import sitemap from '@astrojs/sitemap';

Reviewer: ```suggestion
    import tailwind from '@astrojs/tailwind';
```
